### PR TITLE
[1107] Added provider store

### PR DESCRIPTION
**Pull request by defong**
### Context
Provider store

### Changes proposed in this pull request
Added provider store

### Guidance to review
Part of the arc for locations

This is the store that will provide temporary storage via redis.

The store will be tied to a provider instead of the user, as the data relates to the `provider` and not the `user`

Boiler plate code

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased main
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: spec/services/stores/shared_examples/store.rb [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+    form_store_keys.each do |store_key|
+      context "when form_store_key is #{store_key}" do
+        let(:form_store_key) { store_key }
+
+        let(:value) { nil }
+
+        if described_class == Stores::UserStore
+          # TECH DEBT: [Stores::UserStore] This is error prone
+          let(:redis_key) { "#{identifier_model.id}_#{form_store_key}" }
```
Comment (defong) [inline spec/services/stores/shared_examples/store.rb:36]: it gives

`1_user`

Whereby 
- `1` is an id number and not related to either an `user` or a `course` or a `provider`
- `user` is a form key not as a identifier

**Comment by defong** [comment]
[Bulk Uploads - CSV Flow](https://trello.com/c/DgfeqeZ3/1107-bulk-uploads-csv-flow)

**Review by avinhurry — APPROVED** [review]

File: spec/services/stores/shared_examples/store.rb (hunk 1 above)
Comment (defong) [reply spec/services/stores/shared_examples/store.rb:36]: https://trello.com/c/DCiFvwcr/398-bug-user-store-identifier-id-is-not-unique

File: app/services/stores/user_store.rb [hunk 2]
```diff
@@ -13,6 +13,7 @@ def store_keys
     private
 
     def identifier_id
+      # TECH DEBT: [Stores::UserStore] This is error prone
       identifier_model.id
```
Comment (darokel) [inline app/services/stores/user_store.rb:17]: Just trying to understand the issue a little more but why isn't this following the same format as the others?

Comment (defong) [reply app/services/stores/user_store.rb:17]: I think it was leftovers.

**Pull request merged**